webapp/actions/post_actions.jsx

```diff
@@ -280,6 +280,45 @@ export function removeReaction(channelId, postId, emojiName) {
     AsyncClient.deleteReaction(channelId, reaction);
 }
 
+const postQueue = [];
+
+export function queuePost(post, doLoadPost, success, error) {
```

Comment:
I think this is a bit complicated. Having trouble understanding how it works.

Maybe a bit of abstraction would be good here.